src/libs/ajax/Billing.js

```diff
@@ -0,0 +1,130 @@
+    const res = await fetchOrchestration(`api/billing/v2/${billingProjectName}/billingAccount`,
+      _.merge(authOpts(), { signal, method: 'DELETE' }))
+    return res
+  },
+
+  updateSpendConfiguration: async ({ billingProjectName, datasetGoogleProject, datasetName }) => {
+    const res = await fetchOrchestration(`api/billing/v2/${billingProjectName}/spendReportConfiguration`,
+      _.mergeAll([
+        authOpts(), { signal, method: 'PUT' },
+        jsonBody({ datasetGoogleProject, datasetName })
+      ]))
+    return res
+  },
+
+  /**
+   * Returns the spend report for the given billing project, from 12 AM on the startDate to 11:59 PM on the endDate (UTC). Spend details by
+   * Workspace are included.
+   *
+   * @param billingProjectName
+   * @param startDate, a string of the format YYYY-MM-DD, representing the start date of the report.
+   * @param endDate a string of the format YYYY-MM-DD, representing the end date of the report.
+   * @returns {Promise<*>}
+   */
+  getSpendReport: async ({ billingProjectName, startDate, endDate }) => {
+    const res = await fetchRawls(
+      `billing/v2/${billingProjectName}/spendReport?${qs.stringify({ startDate, endDate, aggregationKey: 'Workspace~Category' })}&${qs.stringify({ aggregationKey: 'Category' })}`,
+      _.merge(authOpts(), { signal })
+    )
+    return res.json()
+  },
+
+  listProjectUsers: async projectName => {
+    const res = await fetchRawls(`billing/v2/${projectName}/members`, _.merge(authOpts(), { signal }))
+    return res.json()
+  },
+
+  addProjectUser: (projectName, roles, email) => {
+    const addRole = role => fetchRawls(
+      `billing/v2/${projectName}/members/${role}/${encodeURIComponent(email)}`,
+      _.merge(authOpts(), { signal, method: 'PUT' })
+    )
+
+    return Promise.all(_.map(addRole, roles))
+  },
+
+  removeProjectUser: (projectName, roles, email) => {
+    const removeRole = role => fetchRawls(
+      `billing/v2/${projectName}/members/${role}/${encodeURIComponent(email)}`,
+      _.merge(authOpts(), { signal, method: 'DELETE' })
+    )
+
+    return Promise.all(_.map(removeRole, roles))
+  },
+
+  changeUserRoles: async (projectName, email, oldRoles, newRoles) => {
+    const billing = Billing()
+    if (!_.isEqual(oldRoles, newRoles)) {
+      await billing.addProjectUser(projectName, _.difference(newRoles, oldRoles), email)
+      return billing.removeProjectUser(projectName, _.difference(oldRoles, newRoles), email)
+    }
+  },
+
+  listAzureManagedApplications: async (subscriptionId, includeAssignedApplications) => {
+    const response = await fetchBillingProfileManager(
+      `azure/v1/managedApps?azureSubscriptionId=${subscriptionId}&includeAssignedApplications=${includeAssignedApplications}`,
+      _.merge(authOpts(), { signal }))
+    return response.json()
+  }
+})
+
+//TODO: this is a weird util method that calls an api... should be refactored so its not a circular utility
```

Comment:
This TODO can be removed now I believe. Might be worth wrapping it in the Billing module though, so it's clear it makes an API call.

Reply:
Yeah, this is kind of weird function. I think I'd vote for splitting the `canCompute` part of this out, leaving something like `canUseWorkspaceThroughBillingProject` that always makes a request. I think I'll leave that for later though and keep this PR as reorganization.